demisto_sdk/commands/download/downloader.py

```diff
@@ -275,47 +273,25 @@ def download_playbook_yaml(self, playbook_string) -> str:
 
         return playbook_string
 
-    def handle_incidentfield(
-        self, incidentfield_string: str, scripts_mapper: dict
-    ) -> str:
-        # In case the incident field uses a custom script, replace the id value of the script with its name
-        file_json_object = json.loads(incidentfield_string)
-        incidentfield_name = file_json_object.get("name")
-        script = file_json_object.get("script")
-        if incidentfield_name and (
-            incidentfield_name in self.input_files or self.all_custom_content
-        ):
-            if script and script in scripts_mapper:
-                incidentfield_string = incidentfield_string.replace(
-                    script, scripts_mapper[script]
-                )
-
-        return incidentfield_string
-
-    def handle_layout(self, layout_string, scripts_mapper):
-        # In case the layout uses a custom script, replace the id value of the script with its name
-        file_json_object = json.loads(layout_string)
-        layout_name = file_json_object.get("name")
-        if layout_name and (layout_name in self.input_files or self.all_custom_content):
-            for tab in (file_json_object.get("detailsV2") or {}).get("tabs", ()):
-                for section in tab.get("sections", ()):
-                    for item in section.get("items", ()):
-                        script_id = item.get("scriptId")
-                        if script_id and script_id in scripts_mapper:
-                            layout_string = layout_string.replace(
-                                script_id, scripts_mapper[script_id]
-                            )
-
-        return layout_string
-
     @staticmethod
-    def map_script(script_string: str, scripts_mapper: dict) -> dict:
+    def map_script(member_name, script_string: str, scripts_mapper: dict) -> dict:
         script_yml = yaml.load(script_string)
-        script_id = script_yml.get("commonfields").get("id")
+        if member_name.lower().lstrip("/").startswith("playbook"):
+            script_id = script_yml.get("id", "")
+        else:
+            script_id = script_yml.get("commonfields", {}).get("id", "")
         if re.search(UUID_REGEX, script_id):
             scripts_mapper[script_id] = script_yml.get("name")
         return scripts_mapper
 
+    @staticmethod
+    def map_json_content_item(script_string: str, scripts_mapper: dict) -> dict:
+        script_json = json.loads(script_string)
+        script_id = script_json.get("id", "")
+        if re.search(UUID_REGEX, script_id):
+            scripts_mapper[script_id] = script_json.get("name")
```

Comment:
please add a `logger.debug` stating each replacement

Reply:
done. very important to use now that logger is implemented. thanks.
( added in the actual replace line )